config/compiler/image.go

```diff
@@ -19,7 +37,16 @@ func expandImage(name string) string {
 	if err != nil {
 		return name
 	}
-	return reference.WithDefaultTag(ref).String()
+	s := ref.String()
+
+	if !strings.Contains(s, ":") {
+		s = fmt.Sprintf("%s:%s", s, defaultTag)
```

Comment:
is this passing unit tests?  this function should return an image name with no tag, and this block looks like it adds a default tag if one doesn't exit.

Reply:
it looks like there is a `TrimNamed` function in the docker distribution package that we could potentially use here. https://godoc.org/github.com/docker/distribution/reference#TrimNamed